go.mod

```diff
@@ -75,3 +75,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
+
+replace gopkg.in/yaml.v3 => gopkg.in/yaml.v3 v3.0.1
```

Comment:
I think we could also use `exclude` here so that future patch bumps are picked up instead of pinning to `v3.0.1` but either way gets the job done.